src/main/scala/collections/ladder/Ladder.scala

```diff
@@ -0,0 +1,17 @@
+package collections.ladder
+
+object Ladder {
+
+  def main(args: Array[String]): Unit = {
+    val maxStep: Int = 10
+    val listSize: LazyList[Int] = calcWays(maxStep)
+    println(s"Number of possible ways: ${listSize.size}")
+  }
+
+  def calcWays(n: Int): LazyList[Int] = {
+    if (n <= 0) LazyList.empty
```

Comment:
the current solution is based on rolling back to the first index, making 3 calls at each step, should be reworked

Reply:
Changed to using result of previous computations